pkg/reconciler/broker/trigger/trigger.go

```diff
@@ -153,9 +153,18 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, t *eventingv1.Trigger) p
 
 func (r *Reconciler) resolveDeadLetterSink(ctx context.Context, b *eventingv1.Broker, t *eventingv1.Trigger) error {
 	// resolve the trigger's dlq first, fall back to the broker's
-	for _, delivery := range []*eventingduckv1.DeliverySpec{t.Spec.Delivery, b.Spec.Delivery} {
+	if t.Spec.Delivery != nil && t.Spec.Delivery.DeadLetterSink != nil && t.Spec.Delivery.DeadLetterSink.Ref != nil &&
+		t.Spec.Delivery.DeadLetterSink.Ref.Namespace == "" {
+		t.Spec.Delivery.DeadLetterSink.Ref.Namespace = t.GetNamespace()
+	}
+
+	var parent interface{} = t
+	for i, delivery := range []*eventingduckv1.DeliverySpec{t.Spec.Delivery, b.Spec.Delivery} {
```

Comment:
in we we do the resolve and fallback - let's capture that too

Reply:
Brokers seems not to be resolving its DLQ ref in the spec. Maybe[ fixing this issue](https://github.com/knative/eventing/issues/5787) i could change this fallback resolution here right?

Reply:
I think the concept of DLQ has been:
1. X routing component (Channel, Broker) has its own DLQ.
    1. User (possibly a cluster admin) specifies the DLQ for X in its `spec.delivery`.
    2. X is responsible of resolving the URI of the DLQ.
    3. X projects the resolved URI in its status.
2. Y subscriber related object (Subscription, Trigger) declares its own DLQ that should override X's for that subscriber
    1. User (possibly a developer) specifies the DLQ for Y in its `spec.delivery`.
    2. X is responsible of resolving the URI of the DLQ.
    3. X projects the resolved URI in Y's status.
    4. If resolution fails, X will fall back to X's resolved DLQ

So following that logic, I think here we shouldn't resolve again the broker's DLQ. You should use the broker's resolve URI in its status, otherwise fail. 

If this is not the current behavior in brokers then let's fix that there indeed  (👍🏽  on #5787)